wagtail/embeds/migrations/0006_add_embed_hash.py

```diff
@@ -0,0 +1,70 @@
+from django.db import migrations, models
+
+from wagtail.embeds.embeds import get_embed_hash
+
+
+def migrate_forwards(apps, schema_editor):
+    Embed = apps.get_model("wagtailembeds.Embed")
+
+    # based each embed being ~500b, available memory being
+    # to 2MB, and leaving plenty of headway for app itself
+    batch_size = 1500
+
+    # allows small batches to be kept in memory in order
+    # to utilise bulk_update()
+    batch = []
+
+    for embed in Embed.objects.all().only("id", "url", "max_width").iterator():
+
+        embed.hash = get_embed_hash(embed.url, embed.max_width)
+        batch.append(embed)
+
+        if len(batch) == batch_size:
+            # save and reset current batch
+            Embed.objects.bulk_update(batch, ["hash"])
+            batch.clear()
+
+    # save any leftovers
+    if batch:
+        Embed.objects.bulk_update(batch, ["hash"])
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("wagtailembeds", "0005_specify_thumbnail_url_max_length"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="embed",
+            name="hash",
+            field=models.CharField(blank=True, max_length=32),
+        ),
+        migrations.RunPython(migrate_forwards, migrations.RunPython.noop),
+        migrations.AlterField(
+            model_name="embed",
+            name="hash",
+            field=models.CharField(db_index=True, max_length=32, unique=True),
+        ),
+        # MySQL needs max length on the unique together fields.
+        # Drop unique together before alter char to text.
+        migrations.AlterUniqueTogether(
+            name="embed",
+            unique_together=set(),
+        ),
+        migrations.AlterField(
+            model_name="embed",
+            name="url",
+            field=models.TextField(),
+        ),
+        migrations.AlterField(
+            model_name="embed",
+            name="thumbnail_url",
+            field=models.TextField(
+                blank=True,
+                default="",
+            ),
+            preserve_default=False,
+        ),
+    ]
```

Comment:
Has this migration been run in reverse? I have a feeling that might throw up some issues. Django does recommend [not including schema changes and RunPython actions in the same migration](https://django.readthedocs.io/en/latest/ref/migration-operations.html#runpython), so may be safer to err on the side of cautiom here. Excerpt below:

> On databases that do support DDL transactions (SQLite and PostgreSQL), RunPython operations do not have any transactions automatically added besides the transactions created for each migration. Thus, on PostgreSQL, for example, you should avoid combining schema changes and RunPython operations in the same migration or you may hit errors like OperationalError: cannot ALTER TABLE "mytable" because it has pending trigger events.